library/cohesity_vmware_job.py

```diff
@@ -0,0 +1,472 @@
+                    "Please make sure the start time is in specified format", module)
+            time_of_day = TimeOfDay()
+            time_of_day.hour = int(start_time[0])
+            time_of_day.minute = int(start_time[1])
+            request_body.start_time = time_of_day
+        request_body.parent_source_id =\
+            get_parent_source_id(module, module.params.get('protection_source'))
+        request_body.source_ids = [request_body.parent_source_id]
+        vmware_source_object_tree = get_vmware_source_objects(module, request_body.parent_source_id)
+        if module.params.get('include'):
+            request_body.source_ids =\
+                get_vmware_object_ids(module, vmware_source_object_tree, module.params.get('include'))
+        if module.params.get('exclude'):
+            request_body.exclude_source_ids =\
+                get_vmware_object_ids(module, vmware_source_object_tree, module.params.get('exclude'))
+        protection_job = cohesity_client.protection_jobs.create_protection_job(request_body)
+        return protection_job
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def delete_vmware_job(module, job_id):
+    '''
+    :param module: object that holds parameters passed to the module
+    :param job_id: protection job id
+    :return:
+    '''
+    try:
+        delete_job_params = DeleteProtectionJobParam()
+        delete_job_params.delete_snapshots = module.params.get('delete_snapshots')
+        cohesity_client.protection_jobs.delete_protection_job(job_id, body=delete_job_params)
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_cohesity_client(module):
+    '''
+    function to get cohesity cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        cluster_vip = module.params.get('cluster')
+        username = module.params.get('username')
+        password = module.params.get('password')
+        domain = 'LOCAL'
+        if "@" in username:
+            user_domain = username.split("@")
+            username = user_domain[0]
+            domain = user_domain[1]
+        cohesity_client = CohesityClient(cluster_vip=cluster_vip,
+                                         username=username,
+                                         password=password,
+                                         domain=domain)
+        return cohesity_client
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def main():
+    argument_spec = dict(
+            cluster=dict(type='str', required=True),
```

Comment:
if required is true and not given, ansible throws a user friendly error ?

Reply:
yes